editor/plugins/control_editor_plugin.h

```diff
@@ -36,6 +36,7 @@
 #include "scene/gui/box_container.h"
 #include "scene/gui/button.h"
 #include "scene/gui/check_box.h"
+#include "scene/gui/check_button.h"
```

Comment:
This should be forward declared.

Reply:
I assume you mean removing the include and inserting `class CheckButton;` into the header? It does not seem to work:

```
editor\plugins\control_editor_plugin.cpp(656): error C2027: use of undefined type 'CheckButton'
B:\Godot\editor\plugins\control_editor_plugin.h(48): note: see declaration of 'CheckButton'
editor\plugins\control_editor_plugin.cpp(664): error C2027: use of undefined type 'CheckButton'
B:\Godot\editor\plugins\control_editor_plugin.h(48): note: see declaration of 'CheckButton'
editor\plugins\control_editor_plugin.cpp(673): error C2027: use of undefined type 'CheckButton'
B:\Godot\editor\plugins\control_editor_plugin.h(48): note: see declaration of 'CheckButton'
editor\plugins\control_editor_plugin.cpp(675): error C2027: use of undefined type 'CheckButton'
B:\Godot\editor\plugins\control_editor_plugin.h(48): note: see declaration of 'CheckButton'
...
```

Reply:
You also need to add the include in the `.cpp`.

The idea with forward-declaring in the header is to reduce bringing dependencies around in places that include that header just for the class definition. If a dependency is only needed in the implementation, it can be forward-declared and included in the .cpp.

Reply:
Ahh okay, thank you for explaining! :pray: